shared/src/main/java/com/couchbase/lite/DocumentFlags.java

```diff
@@ -0,0 +1,47 @@
+//
+// DocumentFlags.java
+//
+// Copyright (c) 2019 Couchbase, Inc All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package com.couchbase.lite;
+
+import com.couchbase.litecore.C4Constants;
+
+public class DocumentFlags {
```

Comment:
Can this be implemented as a kind of bitwise enum? I remembered that Java has EnumSet but I'm not sure if it's appropriate for this or not.

@borrrden I think you have looked into Java Bitwise enum before. What is your opinion on Java implementation?

Reply:
It looks like the standard practice is to use `EnumSet` to accomplish this -> https://stackoverflow.com/questions/6073347/best-practice-of-using-flags-in-java-method